src/datasets.c

```diff
@@ -518,13 +517,16 @@ static int DatasetLoadString(Dataset *set)
         if (r == NULL) {
             line[strlen(line) - 1] = '\0';
             SCLogDebug("line: '%s'", line);
-
-            // coverity[alloc_strlen : FALSE]
             uint8_t decoded[strlen(line)];
-            uint32_t consumed = 0, num_decoded = 0;
-            Base64Ecode code = DecodeBase64(decoded, (uint32_t)strlen(line), (const uint8_t *)line,
-                    (uint32_t)strlen(line), &consumed, &num_decoded, Base64ModeStrict);
-            if (code == BASE64_ECODE_ERR) {
+            uint32_t num_decoded = 0;
+            bool b64d = Base64Decode((const uint8_t *)line, strlen(line), 0, Base64ModeStrict,
+                    decoded, &num_decoded);
+            if (!b64d) {
+                FatalErrorOnInit("bad base64 encoding %s/%s", set->name, set->load);
+                continue;
+            }
+            DEBUG_VALIDATE_BUG_ON(num_decoded >= strlen(line));
+            if (num_decoded == 0 && strlen(line) > 0) {
                 FatalErrorOnInit("bad base64 encoding %s/%s", set->name, set->load);
```

Comment:
Is this reachable ?

Reply:
Should be, yes. Datasets have strict decoding so in case there's a non base64 alphabet in the encoded string, number of decoded bytes should be 0 indicating there was likely bad base64 encoding..

Reply:
But in this case, is not `b64d` false ?

Reply:
Indeed I had overlooked it. On your suggestion, in the next rev, I'll be changing the retval of the fn to return the number of decoded bytes, then, these checks would be valid.